Source/Csla.Blazor/BusinessRulesValidationMessages.razor.cs

```diff
@@ -0,0 +1,166 @@
+		{
+			// Check that the required parameters have been provided
+			if (CurrentEditContext == null)
+			{
+				throw new InvalidOperationException($"{GetType()} requires a cascading parameter " +
+					$"of type {nameof(EditContext)}. For example, you can use {GetType()} inside " +
+					$"an {nameof(EditForm)}.");
+			}
+
+			if (string.IsNullOrWhiteSpace(PropertyName))
+			{
+				throw new InvalidOperationException($"{GetType()} requires a value for the " +
+					$"{nameof(PropertyName)} parameter.");
+			}
+
+			// Wire up handling of the state change events we need event
+			if (CurrentEditContext != _previousEditContext)
+			{
+				DetachPreviousEventHandlers();
+				CurrentEditContext.OnFieldChanged += _fieldChangedHandler;
+				CurrentEditContext.OnValidationStateChanged += _validationStateChangedHandler;
+				_previousEditContext = CurrentEditContext;
+			}
+		}
+
+		protected void OnFieldChanged(object sender, FieldChangedEventArgs eventArgs)
+		{
+			if (eventArgs.FieldIdentifier.FieldName.Equals(PropertyName, StringComparison.InvariantCultureIgnoreCase))
+			{
+				_validationInitiated = true;
+				StateHasChanged();
+			}
+		}
+
+		protected void OnValidationStateChanged(object sender, ValidationStateChangedEventArgs eventArgs)
+		{
+			IEnumerable<string> messages;
+
+			// Retrieve the messages for the property in which we are interested
+			messages = CurrentEditContext.GetValidationMessages(CurrentEditContext.Field(PropertyName));
+
+			if (messages.Any())
+			{
+				// If any messages are present then validation has been run
+				// This is a cheat used to identify that the form has been submitted with invalid data
+				_validationInitiated = true;
+			}
+
+			StateHasChanged();
+		}
+
+		protected void DetachPreviousEventHandlers()
+		{
+			if (_previousEditContext != null)
+			{
+				// Unhook any event handler made to a different edit context
+				_previousEditContext.OnFieldChanged -= _fieldChangedHandler;
+				_previousEditContext.OnValidationStateChanged -= _validationStateChangedHandler;
+			}
+		}
+
+		#endregion
+
+		#region Message Retrieval
+
+		protected IEnumerable<string> GetErrorMessages()
+		{
+			return GetBrokenRuleMessages(PropertyName, RuleSeverity.Error);
+		}
+
+		protected IEnumerable<string> GetWarningMessages()
+		{
+			return GetBrokenRuleMessages(PropertyName, RuleSeverity.Warning);
+		}
+
+		protected IEnumerable<string> GetInfoMessages()
+		{
+			return GetBrokenRuleMessages(PropertyName, RuleSeverity.Information);
+		}
+
+		private IEnumerable<string> GetBrokenRuleMessages(string propertyName, RuleSeverity severity)
+		{
+			IList<string> messages = new List<string>();
+			ICheckRules objectUnderTest;
+
+			// Attempt to gain access to the underlying CSLA object
+			objectUnderTest = CurrentEditContext.Model as ICheckRules;
+			if (objectUnderTest == null)
+			{
+				throw new ArgumentException("Model is not available, or does not implement ICheckRules");
```

Comment:
This string needs to come from a resource so it can be localized. I'll find an example to follow in this regard, as (iirc) we have all the resource strings flowing from `Csla.dll` to keep localization simple for everyone.

Reply:
Yes, agreed.

CSLA is far better than .NET Core in this regard; all of the error messages in the built-in Blazor components seem to be hard-coded at the moment!

Reply:
Here's an example of pulling a resource string from `Csla.dll` in `Csla.Xaml` - so the technique should be the same for `Csla.Blazor`

https://github.com/MarimerLLC/csla/blob/master/Source/Csla.Xaml.Shared/TriggerAction.cs#L87

Reply:
Hehehe. Well, I could make up a few translations, for a laugh ...

Hmm, despite knowing a bit of French and German, colloquialisms can always catch out the inexperienced, so US English it is.

Reply:
As an aside, for a conversation far in the future, it might be that _exception_ messages should not be in the user's language.

On the assumption that you shouldn't show exception details to an end user - because they will both confuse the non-technical, and inform the technical and mischievous, the destination for the exception is the logging system, for reading by the people who run the system, not use the system.

Developers are, in general, likely to be able to speak English. Class names are in English, and programming language features remain in English no matter where people are (I think?) As a result, the people who are looking in the logging system of a multilingual system, might be more likely to benefit from English than from the language of the _user_.

I'm not disagreeing that I should do this - I will do it, for consistency if nothing else - but it makes you think.

For better or worse, our sites are not multilingual - I asked that question, and was told not to make them so. But if they were, a Russian or Greek error message in the logging system - generated because the user's culture was one of those when they hit the website - would be of no help to me at all!

I think that's one to think about for CSLA 6.0. Or maybe 7.0. ;)

Reply:
I suppose, to complete this line of random thought at an inappropriate moment just before a major release, I should say that we are talking about there being two languages of importance, and that both of the languages we are talking about are represented in the system in which the application is executing.

From memory, I think the user's language is represented by CurrentCulture.UICulture. I think the developer's culture is fairly likely to be represented by the culture in which the base operating system was installed. However, in a cloud-based world, even _that_ isn't guaranteed!

Language is _hard_ - ridiculously hard. If only everyone spoke CSLALanguage. Ah, now _there's_ an idea ...

Reply:
I've temporarily used the simpler solution, but raised another issue to add to the resources file for this and several other error messages.